### PR TITLE
first try on adding Docker to help development

### LINKED ISSUE #102: Docker to help development
I added a pull request for adding a dockerfile and docker-compose to the project to help people testing and developing.

### PR DESCRIPTION
**Pull request by mikkeschiren**
Fixes #102

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM node:10.19.0-stretch
```
Comment (joeyciechanowicz) [inline Dockerfile]: You might want to use `[buildkite/puppeteer`](https://hub.docker.com/r/buildkite/puppeteer/tags/)` as I seem to recall it's a real pain getting the `node` image to run Puppeteer. (which Pa11y needs)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    build:
+      context: .
+    ports: 
+      - 8082:8080
+    environment:
+      DATABASE_URL: postgres://sidekick:sidekick@db:5432/sidekick
+      LOG_LEVEL: debug
+      NODE_ENV: development
+      PORT: 8080
+      SESSION_SECRET: b3c542a1-79ce-46a1-b958-06e352eba249
```
Comment (joeyciechanowicz) [inline docker-compose.yml]: Can this be read from the environment variables, as it should be different per installation. Think you can read in variables from the host shell using
```
      SESSION_SECRET: ${SESSION_SECRET}
```

**Review by joeyciechanowicz — COMMENTED** [review]
Great idea adding this, thankyou! Just a couple of comments

File: docker-compose.yml (hunk 2 above)
Comment (mikkeschiren) [reply docker-compose.yml]: Yeah I guess that would be the best.

File: Dockerfile (hunk 1 above)
Comment (mikkeschiren) [reply Dockerfile]: True, puppeter is needed, but did a try on buildkite/puppeteer - but it did not work out with the sidekick code is it is right now - but I will look into it -I will be back on the subject.

Comment (mikkeschiren) [reply Dockerfile]: I updated Dockerfile with another setup that I am using with puppeteer, using alpine.

**Comment by mikkeschiren** [comment]
Updated documentation, fixed puppeteer-core into the image.

Example used for testing puppeteer in the container:

```
const puppeteer = require('puppeteer-core');
    puppeteer
      .launch({
        executablePath: '/usr/bin/chromium-browser',
        args: ['--no-sandbox', '--disable-dev-shm-usage'],
      })
      .then(async (browser) => {
         const page = await browser.newPage();
         await page.goto('https://www.thefarside.com/');
         await page.screenshot({path: 'farside.png'});
         await browser.close();

      });

```

File: Dockerfile (hunk 1 above)
Comment (mikkeschiren) [reply Dockerfile]: Or rather - using puppeteer-core

**Comment by mikkeschiren** [comment]
Just discovered that I broke something i sidekick when I rebuilt the container... I will be back if not someone else do it for me :-)

**Pull request closed without merging**